### ISSUE TITLE
QuPath v0.2.0-m2 Deletes old qpdata files if saving in new version

### ISSUE THREAD

**Issue by NewQuPathUser** [qupath/qupath#292]
Hi everyone,

I tried using QuPath v0.2.0-m2 on my Mac with old qpdata files I had generated. They opened fine and it appeared that I could save the files when I was using the new version with qpdata files made with the old one. However, although it seemed to save, the original qpdata file is deleted or renamed/moved in such a way that I cannot find it anymore. It is not in the trash, and there is not a new/alternate file name generated in the same folder the original file was in.

I know that in the window that appears when you open the new version that it says opening old files can make them incompatible, but I wasn't expecting to have no record of the file anymore. I have some alterations I made that would be great to recover, but otherwise, this is more reporting for a general notice. Thank you.

**Comment by petebankhead** [comment]
There are warnings on startup and at the [bottom of the info page](https://qupath.github.io/QuPath-v0.2.0)...

It's not clear from your description did you open a project or work with .qpdata files directly?

**Comment by Svidro** [comment]
I was not able to replicate this on a Win7 64bit version. Opening either a qpproj file or the qpdata file directly and saving worked in both cases. I was actually somewhat surprised that it did not create a folder, but seemed to save as normal for 0.1.2. 

The only modifications I made to the file was adding annotations. I am not sure if other alterations might result in different Save behavior.

**Comment by petebankhead** [comment]
I also couldn't replicate it. Projects have changed quite thoroughly so the existence of problems doesn't really surprise me (hence the warnings) - but I don't know what specifically has happened in this case.

**Comment by vipulnj** [comment]
After trying v0.2.0-m2 on both Windows and Linux, I saw observed on:
#### WINDOWS
- While working directly with .qpdata files, it behaves as expected. Which means, I could open an image, make annotations, save, close Qupath ... Reopen Qupath, go to File in the menu bar, click open, locate the .qpdata file and select the qpdata file, select the image file and once it is loaded, I can see the previously saved annotations. **(Expected behavior, no issues here)**
_I did not test Projects on the Windows version_

#### LINUX
- I tried doing the same on Linux but lost data (both saved on disk as .qpdata and the annotations on the application which I was about to save) :
  - Clicking Save on the linux version of QuPath v0.2.0-m2 after some changes (or even without changes) deletes the original qpdata file itself ( by original qpdata file, I mean the .qpdata file I moved from windows machine for testing, along with the corresponding image) **(issue)**
  - There is no way I could reopen existing annotations which were made purely using the Linux version because no annotations were saved in a qpdata file in the first place. **(issue)**
  - When I use Projects, I can save and reopen the annotations purely made using the Linux version. **(Expected behavior, no issues here)**

Hopefully that helps.

Thank you! :)

**Comment by petebankhead** [comment]
@vipulnj Thanks very much for this, I think I understand the problem now and have a fix. I will test it later.

In the meantime, if you really do need to write a .qpdata file on Linux outside a project, then after the initial failed attempt running this script should succeed:
```groovy
import qupath.lib.io.PathIO

def imageData = getCurrentImageData()

// Choose a location (here, the one QuPath thinks it wrote to... but didn't)
file = new File(imageData.lastSavedPath)

// Write directly to a stream
file.withOutputStream {
    PathIO.writeImageData(it, imageData)
}

// Check we do have a file now
print file
print file.exists()
```

**Comment by petebankhead** [comment]
This should be fixed in [v0.2.0-m3](https://github.com/qupath/qupath/releases/tag/v0.2.0-m3) - please reopen if it isn't!
